### PR TITLE
Devise setup

### PR DESCRIPTION
**Pull request by Gegardus**
In this part of the project were installed Devise gem and implemented the setup, so that:

- Can register a new user
- User logs in with a combination of email and password
- Hashed passwords should be stored in the database
- Ask for confirmation of email
- Can reset password.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,20 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  # def current_user
+  #   User.first
+  # end
```
Comment (arslanbisharat) [inline app/controllers/application_controller.rb]: - [x] Let's remove the commented code. This makes it hard to read the code.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+  protect_from_forgery with: :exception
+
+  # prevent users to perfrom action without authentication
+  # before_action :authenticate_user
```
Comment (arslanbisharat) [inline app/controllers/application_controller.rb]: - [x] Let;s remove commented code.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (arslanbisharat) [inline app/models/user.rb]: - [x] Please make sure `:confirmable` is enabled in the user model

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Gegardus) [reply app/controllers/application_controller.rb]: Thank you. Done.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (Gegardus) [reply app/controllers/application_controller.rb]: Thank you. Done.

File: app/models/user.rb (hunk 3 above)
Comment (Gegardus) [reply app/models/user.rb]: Thank you. Done.

**Review by elmar8287 — APPROVED** [review]
Hi, 

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gegardus** [comment]
Thank you @elmar8287

**Pull request merged**